File: js-quest06/ex00/my_string_index.js
/*
**
** QWASAR.IO -- my_string_index
**
**
** @param {String} haystack
** @param {Character} needle
** @return {integer}
**
*/

function my_string_index(haystack, needle) {
    var i = 0;
    while (i < haystack.length) {
        if (haystack[i] == needle) {
            return i;
        } 
        i++;
    }
    return -1;
}

// console.log(my_string_index("hello", "l"));
// console.log(my_string_index("aaaaa", "b"));

File: c-quest02/ex08/my_string_index.c
#include <stdio.h>

/*
**
** QWASAR.IO -- my_string_index
**
** @param {char*} haystack
** @param {char} needle
**
** @return {int}
**
*/

int my_string_index(char* haystack, char needle)
{
    int i;
    i = 0;

    while (haystack[i] != '\0') {
        if (haystack[i] == needle) {
            return i;
        }
        i++;
    }
    return -1;
}